src/topology/continuous_function/units.lean

```diff
@@ -0,0 +1,90 @@
+/-
+Copyright (c) 2022 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+import topology.continuous_function.compact
+import analysis.normed_space.units
+import algebra.algebra.spectrum
+
+/-!
+# Units of continuous functions
+
+This file concerns itself with `C(α, β)ˣ` and `C(α, βˣ)` when `α` is a topological space
+and `β` has an appropriate algebraic and topological structure.
+-/
+
+variables {α β : Type*} [topological_space α]
+
+namespace continuous_map
+
+section monoid
+
+variables [monoid β] [topological_space β] [has_continuous_mul β]
+
+/-- Equivalence between continuous maps into the units of a monoid with continuous multiplication
+and the units of the monoid of continuous maps. -/
+@[to_additive, simps]
+def units_lift : C(α, βˣ) ≃ C(α, β)ˣ :=
+{ to_fun := λ f,
+  { val := ⟨λ x, f x, units.continuous_coe.comp f.continuous⟩,
+    inv := ⟨λ x, ↑(f x)⁻¹, units.continuous_coe.comp (continuous_inv.comp f.continuous)⟩,
+    val_inv := ext $ λ x, units.mul_inv _,
+    inv_val := ext $ λ x, units.inv_mul _ },
+  inv_fun := λ f,
+  { to_fun := λ x, ⟨f x, f⁻¹ x, continuous_map.congr_fun f.mul_inv x,
+                                continuous_map.congr_fun f.inv_mul x⟩,
+    continuous_to_fun := continuous_induced_rng $ continuous.prod_mk (f : C(α, β)).continuous
+      $ mul_opposite.continuous_op.comp (↑f⁻¹ : C(α, β)).continuous },
+  left_inv := λ f, by { ext, refl },
+  right_inv := λ f, by { ext, refl } }
+
+end monoid
+
+section normed_ring
+
+variables [normed_ring β] [complete_space β]
+
+/-- Construct a continuous map into the group of units of a normed ring from a function into the
+normed ring and a proof that every element of the range is a unit. -/
+@[simps]
+noncomputable def units_of_forall_is_unit {f : C(α, β)} (h : ∀ x, is_unit (f x)) : C(α, βˣ) :=
+{ to_fun := λ x, (h x).unit,
+  continuous_to_fun :=
+  begin
+    refine continuous_induced_rng (continuous.prod_mk f.continuous
+      (mul_opposite.continuous_op.comp (continuous_iff_continuous_at.mpr (λ x, _)))),
+    have := normed_ring.inverse_continuous_at (h x).unit,
+    simp only [←ring.inverse_unit, is_unit.unit_spec, ←function.comp_apply] at this ⊢,
+    exact this.comp (f.continuous_at x),
+  end }
```

Comment:
Is there a `normed_ring.is_unit_unit_continuous_at` lemma that makes sense to extract here?

Reply:
I'm not sure exactly what you mean? Do you mean this?
```lean
lemma _root_.normed_ring.is_unit_unit_continuous_at {f : α → β} (h : ∀ x, is_unit (f x))
  (a : α) : continuous_at (λ x, (h x).unit) a := sorry
```
If so, no, I don't think it's worth it because people should just use the `def`. If that's not what you meant, then I'm not sure what function you had in mind to say is `continuous_at`.

Reply:
I was thinking something more like:
```lean
lemma _root_.normed_ring.is_unit_unit_continuous_at {f : α → β} (a : α) (hf : continuous_at f a) (h : is_unit (f a)) :
  continuous_at (λ x, hx.unit) a := sorry
```
assuming its true.

Reply:
Oh I guess that doesn't make sense, so yes, I mean your lemma.

Reply:
I extracted this as a lemma:
```lean
lemma _root_.normed_ring.is_unit_unit_continuous {f : C(α, β)} (h : ∀ x, is_unit (f x)) : 
  continuous (λ x, (h x).unit) := sorry
```
Given our API around induced topologies, I think it's actually easier (and maybe more useful?) to just prove the whole function is continuous, rather than just continuous at a point.

Reply:
The `continuous_at` version feels more general to me though, unless I'm missing something